### DISCUSSION TITLE
vwc-expansion-panel / vwc-expansion-panel-list

### DISCUSSION THREAD

**Discussion by lagjamhero** [Vonage/vivid#740] [New Developments]
**Topic 1**
```js
<vwc-expansion-panel-list>
  <vwc-expansion-panel header="Item 1" open>Some content</vwc-expansion-panel>
  <vwc-expansion-panel header="Item 2" open>Some content</vwc-expansion-panel>
  <vwc-expansion-panel header="Item 3" open>Some content</vwc-expansion-panel>
</vwc-expansion-panel-list>
```

There are scenarios where a page may require all expansion panels to be open on render.
Currently `vwc-expansion-panel` supports an `open` attribute which can be placed on each item.
Alternatively we could also expose an `openAll` prop on `vwc-expansion-panel-list`, but this could be a conflict.

**Topic 2**
```js
<vwc-expansion-panel header="Item 1">Some content</vwc-expansion-panel>
<vwc-expansion-panel header="Item 1" chevronToggle>Some content</vwc-expansion-panel>
```
By design our default toggle icons for [expansion panel](https://vivid.vonage.com/?path=/story/components-atoms-expansion-panel--basic) are +/-
<img width="645" alt="Screenshot 2021-04-02 at 09 54 20" src="https://user-images.githubusercontent.com/10006963/113352939-72309d00-9399-11eb-816a-865a01cdf525.png">

We also have a requirement to support chevron toggles
<img width="642" alt="Screenshot 2021-04-02 at 09 55 28" src="https://user-images.githubusercontent.com/10006963/113353036-98eed380-9399-11eb-8015-8a708a0852aa.png">

currently `chevronToggle` boolean is being used. we don't have designs of more toggle types but maybe a string `toggleIcon` might be useful.
Please **comment your thoughts** on suitable prop names. 

mwc-tabs have an `indicatorIcon` prop that takes an icon and layers over the tab icon when active, rather than placing a line below. Slightly different use case but dropping here too for ideas.

`indicatoricon="donut_large"`
<img width="51" alt="Screenshot 2021-04-02 at 10 03 56" src="https://user-images.githubusercontent.com/10006963/113353886-d011b480-939a-11eb-9d5e-6402773a9b5d.png">
<img width="69" alt="Screenshot 2021-04-02 at 10 04 15" src="https://user-images.githubusercontent.com/10006963/113353890-d142e180-939a-11eb-9032-267df16c83c3.png">

**Comment by yinonov** [comment]
more issues noticed

- icon property overrides the indicator button unless setting trailingToggle attribute

Expansion panel **list** PR can be found [here](https://github.com/Vonage/vivid/pull/683)

**Reply by lagjamhero** [reply]
addressed in commit 040a733

**Comment by k-paxian** [comment]
I would suggest adding capability to lazily render panel content, when opened only. This will save us some performance & [issues](https://github.com/Vonage/vivid/issues/789) with other components rendered when panel is closed & need to update/realign/relayout the content when panel is opened.

**Reply by yinonov** [reply]
sure, considering [template](https://www.w3schools.com/tags/tag_template.asp) and [containment](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Containment). any other suggestions alike? or other ideas?

**Reply by yinonov** [reply]
added to ticket https://jira.vonage.com/browse/VIV-519

**Comment by yinonov** [comment]
suggesting spec for part 1 -
openAll seems like a rare case - don't feature. if anyway consumer wants to open all expansion panel it kind of deviates from the purpose of wrapping it with accordion in the 1st place and each expansion panel should be set as open individually.

alternatively it's possible to follow [angular's practice](https://material.angular.io/components/expansion/overview#accordion) with multiple prop but again, not sure why to wrap with accordion...

**Reply by lagjamhero** [reply]
Removed `openAll` in latest commits. 

Dev can use alternative suggestion of `multi` combined with multiple `open` props if needed.

I think `multi` is useful on it's own and can be used in navigation scenarios

**Comment by yinonov** [comment]
part 2 suggestion -
`indicator-icon-set="chevron | binary"` this enables enhancement at a later point to add more sets

@JoelGraham93 
edit: add 'none' to the predefined options

**Reply by lagjamhero** [reply]
added suggestion to PR

**Comment by yinonov** [comment]
I think it makes more sense to default to `trailingToggle=true` or even better change to 'leadingIcon=false'. This is more common
